hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/utils/HddsVolumeUtil.java

```diff
@@ -215,6 +227,17 @@ public static boolean checkVolume(HddsVolume hddsVolume, String scmId,
         logger.error("Unable to create ID directory {} for datanode.", idDir);
         return false;
       }
+
+      if (VersionedDatanodeFeatures.SchemaV3.chooseSchemaVersion()
+          .equals(OzoneConsts.SCHEMA_V3)) {
```

Comment:
Add a check that hddsVolume.setDbParentDir is set to avoid duplicate rockdb initialization.

Reply:
Actually we don't have to, because here we have the check above that `hddsFiles.length == 1`, so we are sure that this `HddsVolume` is fresh.